genesis/example/allowance.go

```diff
@@ -29,13 +29,16 @@ type Allowance interface {
 	contract.SmartContract
 	GetAmount() int
 	GetSender() string
+	GetReciever() string
+	MarkCompleted()
```

Comment:
You probably need `IsCompleted()` method as well, so we can check only not completed allowances later for getting balance.